tests/k8s_otel/kind_wrapper.py

```diff
@@ -0,0 +1,215 @@
+import time
+import random
+import subprocess
+import socket
+import docker
+import os
+from docker import DockerClient
+from pytest_kind import KindCluster
+from subprocess import CalledProcessError
+from contextlib import contextmanager
+from typing import Generator
+
+from jina.logging.logger import JinaLogger
+
+
+class KindClusterWrapperV2:
```

Comment:
Why is this needed? what is it doing that we were not doing before in other tests?

Reply:
Hrmm. It does things we were doing in other tests in a more consolidated and simpler fashion. Most notably the following:

**Deploying artifacts from a directory**
Other tests: https://github.com/jina-ai/jina/blob/master/tests/k8s/test_k8s.py#L26-L121
This test: https://github.com/jina-ai/jina/pull/5372/files#diff-22c1d4e56c2c85e03d9fca28239f7aa37559fd8d449f8629c73135afcbb654c4R57-R82

1. The old method requires you to know the exact names of your deployments and supply them in `deployment_replicas_expected`. The new one infers them and thus does not need this argument.
2. The old method uses `api_client`, `app_client` and `core_client` from python kubernetes as arguments. This creates an unnecessary dependency on the python implemention of kubernetes client when these methods can simply be performed with kubectl. Despite having all the clients, the old method still needs a kubernetes utils inside itself.
3. If the deployment were to fail, the old method provides no observability into the state of the cluster.

**Linkerd**
Linkerd is installed unnecessarily in the old tests.
1. This create issues with installing it, as mentioned here: https://github.com/jina-ai/jina/blob/master/tests/k8s/conftest.py#L69-L70.
2. Installing Linkerd forces all tests and development of the tests to need to work inside its service mesh framework which adds about a few minutes to debug operations that could have been done in 5 seconds because more time was spent on the overhead of figuring out why the service mesh isn't happy with your cert than actually looking at your containers.  

If there is some linkerd integration that needs to be tested, they should not be tested with the other k8s test. Though as far as I am aware, everything would work just fine without it. The only place other than installing itself in which it is referred is https://github.com/jina-ai/jina/blob/master/tests/unit/orchestrate/deployments/config/test_k8s_deployment_config.py#L339. Which imo, is more a test that linkerd works than a test that jina works with it.

Reply:
then why do not u just change all the tests into using this? 

I do not want this much duplication, let's choose 1 way and only 1 to do these kind of tests

Reply:
Yep agreed. So I port the old ones to use this in another PR?